File: frontend/src/api/datax-task.ts
import request from './request'
import type { ApiResult } from '@/types'

export interface FieldMapping {
  readerColumn: string
  writerColumn: string
  targetType: string
  maskingRuleId: number | null
}

export interface DataxTask {
  id: number
  name: string
  description?: string
  readerDatasourceId?: number | null
  readerDatabase?: string
  readerTable?: string
  readerColumns?: string[]
  readerWhere?: string
  writerDatasourceId?: number | null
  writerDatabase?: string
  writerTable?: string
  writerColumns?: string[]
  writeMode?: string
  fieldMappings?: FieldMapping[]
  channel?: number
  splitPk?: string
  batchSize?: number
  encoding?: string
  preSql?: string
  postSql?: string
  configMode?: string
  scriptContent?: string
  status?: string
  createdAt: string
  updatedAt: string
}

export interface DataxTaskRequest {
  id?: number
  name: string
  description?: string
  readerDatasourceId?: number | null
  readerDatabase?: string
  readerTable?: string
  readerColumns?: string[]
  readerWhere?: string
  writerDatasourceId?: number | null
  writerDatabase?: string
  writerTable?: string
  writerColumns?: string[]
  writeMode?: string
  fieldMappings?: FieldMapping[]
  channel?: number
  splitPk?: string
  batchSize?: number
  encoding?: string
  preSql?: string
  postSql?: string
  configMode?: string
  scriptContent?: string
}

export function listDataxTasks(): Promise<ApiResult<DataxTask[]>> {
  return request.get('/datax-task/list')
}

export function getDataxTask(id: number): Promise<ApiResult<DataxTask>> {
  return request.get(`/datax-task/${id}`)
}

export function createDataxTask(data: DataxTaskRequest): Promise<ApiResult<DataxTask>> {
  return request.post('/datax-task/create', data)
}

export function updateDataxTask(data: DataxTaskRequest): Promise<ApiResult<DataxTask>> {
  return request.put('/datax-task/update', data)
}

export function deleteDataxTask(id: number): Promise<ApiResult<null>> {
  return request.delete(`/datax-task/${id}`)
}

export function publishDataxTask(id: number): Promise<ApiResult<DataxTask>> {
  return request.post(`/datax-task/${id}/publish`)
}

export function unpublishDataxTask(id: number): Promise<ApiResult<DataxTask>> {
  return request.post(`/datax-task/${id}/unpublish`)
}

export function getDataxTaskJson(id: number): Promise<ApiResult<{ dataxJson: string }>> {
  return request.get(`/datax-task/${id}/datax-json`)
}

File: backend/src/main/java/com/idata/entity/DataxTask.java
package com.idata.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("datax_task")
public class DataxTask {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    private Long readerDatasourceId;

    private String readerDatabase;

    private String readerTable;

    private String readerColumns;

    private String readerWhere;

    private Long writerDatasourceId;

    private String writerDatabase;

    private String writerTable;

    private String writerColumns;

    private String writeMode;

    private String fieldMappings;

    private Integer channel;

    private String splitPk;

    private Integer batchSize;

    private String encoding;

    private String preSql;

    private String postSql;

    private String status;

    private String configMode;

    private String scriptContent;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
